guidelines/github-contribution-guide.md

```diff
@@ -0,0 +1,268 @@
+
+```
+git add <file1> <file2>
+```
+
+- You can now create your commit containing the changes you just added.
+- Your commit message should contain the following information:
+  - A one line commit title summarizing the commit, followed by an empty line
+  - A multi-line commit message body explaining the need for the change and how you approached it
+  - A link to GitHub issue (if exists), using syntax like `Resolves #<GitHub issue number>` so that the GitHub issue gets linked to the pull request.
+  - The `-s` flag will automatically sign the commit
+
+```
+git commit -s
+```
+
+- You can now push your local changes to your forked repository
+
+```
+git push origin <feature_branch_name>
+```
+
+Note - If you want to integrate upstream changes from the original repository
+before pushing your changes see [Syncing Your Fork With the Upstream
+Repository](#syncing-your-fork-with-the-upstream-repository).
+
+You have now successfully pushed your local changes to your forked repository.
+To integrate these changes you must now go through the pull request process.
+
+## Opening a Pull Request in GitHub
+
+Now that you've created and pushed changes to a feature branch in your forked repository,
+you can now open a pull request against the original Hyperledger repository from which you created your fork and begin the code review process.
+
+- Navigate to https://github.com/hyperledger/<original_repository>
+- Select the Pull Request tab at the top of the page
+- In the top right corner of the Pull Requests page, select New Pull Request
+- On the Compare Changes page, select "compare across forks" at the top of the page
+- Select the Hyperledger repository and branch that you want to merge into as the "base repository" and "base" branch
+- Select your fork and your feature branch as the "head repository" and "compare" branch
+- Select Create Pull Request
+- You'll have the option to open your pull request as a Draft if you are not yet ready to have it reviewed.
+
+Congratulations, you have now submitted your first pull request to a Hyperledger project.
+Your pull request will now run through CI checks.
+You can monitor your pull request CI progress by navigating to the Checks tab of the pull request.
+If there are CI check failures the pull request may not get reviewed and merged.
+Either resolve the failures by updating the pull request, or add a comment to the pull request to request help resolving the failure.
+
+## Updating a Pull Request
+
+As you receive review comments on your pull request (or if pull request checks fail), you may need to make edits to your commit.
+In the local branch you are working from, you may make changes, amend the commit, and re-push to your remote feature branch.
+
+```
+git add -p
+git commit --amend
+git push origin <feature_branch_name> -f
+```
+
+This will update the pull request and re-trigger checks.
+
+If you have created multiple commits locally, it is typically best to squash them into a single final commit.
```

Comment:
[nitpick] Few of the projects I have contributed expect a cleaner commit history. i.e. The commits shall be feature specific and history should reflect incremental and independent change set.

Reply:
Updating the wording.